src/dialect/dialect.ts

```diff
@@ -19,4 +26,54 @@ export interface Dialect extends KyselyDialect {
     callback?: ColumnBuilderCallback
   ): CreateTableBuilder<TB>;
 
+  /**
+   * This is a helper method to add a column with foreign key constraints.
+   * This is primarily useful because the `mySQL` dialect adds the constraints in a different way than `sqlite` and `postgres`.
+   *
+   * @param builder the CreateTableBuilder to add the column to.
+   * @param tableName the name of the table to add the column to.
+   * @param columnName the name of the column to add.
+   * @param targetType the type of the column to add.
+   * @param referenceTable the foreign table to reference.
+   * @param referenceColumnName the foreign column to reference.
+   * @param onDeleteAction the action to take when the referenced row is deleted.
+   *
+   * The ForeignKey name it creates in `mysql` will be in the following format:
+   * `${referenceTable}_${referenceColumnName}__${tableName}_${columnName}`
+   * ex: if the reference table is `users` and the reference column is `id` and the table is `profiles` and the column is `userId`,
+   * the resulting name for the foreign key is: `users_id__profiles_userId`
```

Comment:
I missed some context here:
1. Should this documentation just be in mysql imple?
1. Why is it that the other 2 SQL dialects don't require this?

Reply:
1. Yeah I'll move this blurb into the `mysql` implementation of the interface.

2. MySQL is a very special child of the SQL world 😞:
`MySQL parses but ignores “inline REFERENCES specifications” (as defined in the SQL standard) where the references are defined as part of the column specification. MySQL accepts REFERENCES clauses only when specified as part of a separate FOREIGN KEY specification.`

https://dev.mysql.com/doc/mysql-reslimits-excerpt/8.0/en/ansi-diff-foreign-keys.html

Reply:
😮